tests/src/com/unciv/testing/BasicTests.kt

```diff
@@ -157,6 +157,21 @@ class BasicTests {
         Assert.assertTrue("This test succeeds only if all uniques of buildings are presented in UniqueType.values()", allOK)
     }
 
+    @Test
+    fun allPromotionsUniquesHaveTheirUniqueTypes() {
+        val promotions = ruleset.unitPromotions.values
+        var allOK = true
+        for (promotion in promotions) {
+            for (unique in promotion.uniques) {
+                if (!UniqueType.values().any { it.placeholderText == unique.getPlaceholderText() }) {
+                    println("${promotion.name}: $unique")
+                    allOK = false
+                }
+            }
+        }
+        Assert.assertTrue("This test succeeds only if all uniques of promotions are presented in UniqueType.values()", allOK)
+    }
```

Comment:
Not sure I agree with this test, as untyped uniques can be used as identifiers by modders

Reply:
I agree with this test, since it's talking about the base game.

Reply:
But sounds more like something for the modchecker? There is already a unit test calling it. Then maybe do a "completeness" check for each and every RulesetObject class, with for the moment differentiated expected results for base and mods? Overall uniquetypification completeness can be done using a debug option already.

Reply:
> Then maybe do a "completeness" check for each and every RulesetObject class

I agree on that. The amount of those types (and types of types) is so large, that new developer will have a hard time to add/edit something there without a chance to mess up the things.